Misc/WithKeyUpdateArgs1.KeyUpdate Failure

### Describe the bug

Recent CI test failure:
```
##[error][06/03/2021 16:20:53] Misc/WithKeyUpdateArgs1.KeyUpdate/3 failed:
Note: Google Test filter = Misc/WithKeyUpdateArgs1.KeyUpdate/3
[==========] Running 1 test from 1 test suite.
[----------] Global test environment set-up.
Initializing for User Mode tests
[----------] 1 test from Misc/WithKeyUpdateArgs1
[ RUN      ] Misc/WithKeyUpdateArgs1.KeyUpdate/3
/home/vsts/work/1/s/src/test/lib/HandshakeTest.cpp:1975: Failure
Server->ForceKeyUpdate() failed, 0x1
[  FAILED  ] Misc/WithKeyUpdateArgs1.KeyUpdate/3, where GetParam() = v4/3 (409 ms)
[----------] 1 test from Misc/WithKeyUpdateArgs1 (409 ms total)
```

https://dev.azure.com/ms/msquic/_build/results?buildId=175190&view=logs&j=d91c8496-843a-5c6a-cc6b-c76b8a319d75&t=fc1fc1de-6dba-547e-ec32-cb816dfc526b&l=1342

Likely need to increase the test code's retry count to be more resilient to this.

--- a/src/test/lib/TestConnection.cpp
+++ b/src/test/lib/TestConnection.cpp
@@ -226,7 +226,7 @@ TestConnection::ForceKeyUpdate()
                 0,
                 nullptr);
 
-    } while (Status == QUIC_STATUS_INVALID_STATE && ++Try <= 3);
+    } while (Status == QUIC_STATUS_INVALID_STATE && ++Try <= 10);
 
     return Status;
 }
@@ -255,7 +255,7 @@ TestConnection::ForceCidUpdate()
                 0,
                 nullptr);
 
-    } while (Status == QUIC_STATUS_INVALID_STATE && ++Try <= 3);
+    } while (Status == QUIC_STATUS_INVALID_STATE && ++Try <= 10);
 
     return Status;
 }

